### ISSUE TITLE
Missing required attributes from merged entities

### ISSUE THREAD

**Issue by cburbank** [ruby-grape/grape-swagger-entity#34]
With #33 exposed attributes are now being correctly merged, but the `required` array is not reflecting the merged attributes.

```ruby
class CourseDirector < Grape::Entity
  expose :title, documentation: { type: 'String', required: true }
  expose :user, merge: true, using: Entities::Faculty
end
```

In the resulting swagger JSON you can see where `CourseDirector.properties` includes the merged properties from `user`, but `CourseDirector.required` does not, and has `user` instead:

![image](https://user-images.githubusercontent.com/15691227/43857735-35875850-9b1a-11e8-991a-6fa38855e6ae.png)

**Comment by Bugagazavr** [comment]
Hello, thank you for your report

But we have some problems with this case, because merge implemented inside this gem, and we need 
 to return pure properties ( not definition ) for top level, but required properties it's a part of definition. It works for nested, properties, because we need anyway to build nested definition any way.

@LeFnord Hello, i guess we can return tuple with hash of properties and array of required attributes for this place https://github.com/ruby-grape/grape-swagger/blob/master/lib/grape-swagger/endpoint.rb#L351, this also might help you to drop this crutches https://github.com/ruby-grape/grape-swagger/blob/master/lib/grape-swagger/doc_methods/build_model_definition.rb#L18-L36

**Comment by Bugagazavr** [comment]
Fixed in grape-swagger 0.31.0 and grape-swagger-entity 0.3.0
